src/main/java/com/telekom/m2m/cot/restsdk/CloudOfThingsRestClient.java

```
@@ -607,47 +607,14 @@ public String doPutRequestWithIdResponse(String data, String path, String conten
 
 
     public void delete(String id, String api) {
-        Request request = new Request.Builder()
-                .addHeader("Authorization", "Basic " + encodedAuthString)
-                .url(host + "/" + trimSlashes(api) + "/" + id)
-                .delete()
-                .build();
-        Response response = null;
-        try {
-            response = client.newCall(request).execute();
-            if (!response.isSuccessful()) {
-                throw new CotSdkException(response.code(), "Error in delete with ID '" + id + "' (see https://http.cat/" + response.code() + ")");
-            }
-        } catch( CotSdkException e) {
-            // We need to rethrow this in order to not loose the status code.
-            throw e;
-        } catch (Exception e) {
-            throw new CotSdkException("Error in request", e);
-        } finally {
-            closeResponseBodyIfResponseAndBodyNotNull(response);
+        if(id == null || id.isEmpty()) {
```

:+1: It's good to fail early.